test/distributed/test_dist_neighbor_loader.py

```diff
@@ -0,0 +1,264 @@
+    assert str(mp.current_process().pid) in str(loader)
+    assert isinstance(loader.neighbor_sampler, DistNeighborSampler)
+    assert part_data[0].meta['is_hetero']
+
+    for batch in loader:
+        assert isinstance(batch, HeteroData)
+        assert batch['v0'].input_id.numel() == batch['v0'].batch_size == 10
+
+        assert len(batch.node_types) == 2
+        for node_type in batch.node_types:
+            assert torch.equal(batch[node_type].x, batch.x_dict[node_type])
+            assert batch.x_dict[node_type].device == device
+            assert batch.x_dict[node_type].size(0) >= 0
+            assert batch[node_type].n_id.size(0) == batch[node_type].num_nodes
+
+        assert len(batch.edge_types) == 4
+        for edge_type in batch.edge_types:
+            assert batch[edge_type].edge_index.device == device
+            assert batch[edge_type].edge_attr.device == device
+            assert (batch[edge_type].edge_attr.size(0) ==
+                    batch[edge_type].edge_index.size(1))
+
+            if batch[edge_type].edge_index.numel() > 0:  # Test edge mapping:
+                src, _, dst = edge_type
+                edge_index = part_data[1]._edge_index[(edge_type, "coo")]
+                global_edge_index_1 = torch.stack([
+                    batch[src].n_id[batch[edge_type].edge_index[0]],
+                    batch[dst].n_id[batch[edge_type].edge_index[1]],
+                ], dim=0)
+                global_edge_index_2 = edge_index[:, batch[edge_type].e_id]
+                assert torch.equal(global_edge_index_1, global_edge_index_2)
+
+
+@onlyLinux
+@withPackage('pyg_lib')
+@pytest.mark.parametrize('num_parts', [2])
+@pytest.mark.parametrize('num_workers', [0])
+@pytest.mark.parametrize('async_sampling', [True])
+def test_dist_neighbor_loader_homo(
+    tmp_path,
+    num_parts,
+    num_workers,
+    async_sampling,
+):
+    mp_context = torch.multiprocessing.get_context('spawn')
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    s.bind(('127.0.0.1', 0))
+    port = s.getsockname()[1]
+    s.close()
+    addr = 'localhost'
+
+    data = FakeDataset(
+        num_graphs=1,
+        avg_num_nodes=100,
+        avg_degree=3,
+        edge_dim=2,
+    )[0]
+    partitioner = Partitioner(data, num_parts, tmp_path)
+    partitioner.generate_partition()
+
+    w0 = mp_context.Process(
+        target=dist_neighbor_loader_homo,
+        args=(tmp_path, num_parts, 0, addr, port, num_workers, async_sampling),
+    )
+
+    w1 = mp_context.Process(
+        target=dist_neighbor_loader_homo,
+        args=(tmp_path, num_parts, 1, addr, port, num_workers, async_sampling),
+    )
+
+    w0.start()
+    w1.start()
+    w0.join()
+    w1.join()
+
+
+@onlyLinux
+@withPackage('pyg_lib')
+@pytest.mark.parametrize('num_parts', [2])
+@pytest.mark.parametrize('num_workers', [0])
+@pytest.mark.parametrize('async_sampling', [True])
+@pytest.mark.skip(reason="Breaks with no attribute 'num_hops'")
```

Comment:
@JakubPietrakIntel This test currently breaks for me. I am disabling it for now.